lib/charms/tls_certificates_interface/v2/tls_certificates.py

```diff
@@ -1174,10 +1181,18 @@ def _provider_certificates(self) -> List[Dict[str, str]]:
         """Returns list of provider CSRs from relation data."""
         relation = self.model.get_relation(self.relationship_name)
         if not relation:
-            raise RuntimeError(f"Relation {self.relationship_name} does not exist")
+            logger.debug("No relation: %s", self.relationship_name)
+            return []
         if not relation.app:
-            raise RuntimeError(f"Remote app for relation {self.relationship_name} does not exist")
+            logger.debug("No remote app in relation: %s", self.relationship_name)
+            return []
         provider_relation_data = _load_relation_data(relation.data[relation.app])
+        if not self._relation_data_is_valid(provider_relation_data):
+            logger.warning(
+                "Provider relation data did not pass JSON Schema validation: %s",
```

Comment:
We probably don't need to specify the content of the relation data in the logs.

Reply:
Agreed, removed